staging/src/k8s.io/controller-manager/pkg/leadermigration/config/config_test.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	internal "k8s.io/controller-manager/config"
+)
+
+func TestReadLeaderMigrationConfiguration(t *testing.T) {
+	testCases := []struct {
+		name      string
+		content   string
+		expected  *internal.LeaderMigrationConfiguration
+		exceptErr bool
+	}{
+		{
+			name:      "empty",
+			content:   "",
+			expected:  nil,
+			exceptErr: true,
+		},
+		{
+			name: "basic",
+			content: `
+apiVersion: controllermanager.config.k8s.io/v1alpha1
+kind: LeaderMigrationConfiguration
+leaderName: migration-120-to-121
+resourceLock: leases
```

Comment:
Please add a test for resourceLock: endoints

Reply:
Added. Sorry for missing that out.